src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,142 @@
+        const [subData] = response.data;
+        const embed: MessageEmbed = this._generateEmbedSub(subData);
+
+        message.reply(embed);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
+  }
+
+  private async _generateEmbedList(): Promise<void> {
+    const lastListUpdate: number = Date.now() - this._LAST_UPD_TIME;
+
+    if (lastListUpdate < this._SUB_UPD_THRESH) {
+      return;
+    }
+
+    this._LAST_UPD_TIME = Date.now(); //since it only updates once a day, don't need to worry about accuracy
+
+    await this._updateList();
+
+    //setting up number of columns and rows for the list
+    const maxCol = 3;
+    const maxRows = 10;
+    const numCols = Math.min(maxCol, Math.ceil(this._SUBS.length / maxRows));
+    const numRows = Math.ceil(this._SUBS.length / numCols);
+
+    //sets up a new embed for the list to reside in
+    this._EMBED_LIST = new MessageEmbed();
+    this._EMBED_LIST.setColor('#0099ff').setTitle('Types of Subs');
+    const subs = this._SUBS.map((sub) => this._normalizeName(sub));
+
+    //cycles through each column inserting their row, and also notes the alphabetic range
+    for (let cols = 0; cols < numCols; cols++) {
+      const columnSubs = subs.slice(numRows * cols, numRows * (cols + 1));
+
+      //each column is a embed field note: only 3 fields can be lined up
+      this._EMBED_LIST.addField(
+        `${columnSubs[0].charAt(0)} - ${columnSubs[columnSubs.length - 1].charAt(0)}`,
+        columnSubs.join('\n'),
+        true
+      );
     }
   }
+
+  private _generateEmbedSub(subData: any): MessageEmbed {
+    const embed: MessageEmbed = new MessageEmbed();
+    const parsedNamed = this._normalizeName(subData.sub_name);
+
+    //defaults to not on sale
+    let saleInfo = `it **isn't** on sale currently :sob:, but who could say no to ${subData.price}.  The last time it was on sale was ${subData.last_sale}`;
+
+    //status is true if the sub is on sale
+    if (subData.status.toLowerCase() === 'true') {
+      saleInfo = `it **is** on sale for ${subData.price} :partying_face:, until ${
+        subData.last_sale.split('-')[1]
+      }`;
+    }
+
+    embed.setColor('#0099ff').setTitle(parsedNamed + ' sub');
+
+    embed.setDescription(
+      `Get your own *${parsedNamed}* sub, ${saleInfo}.` +
+        ` So what are you waiting for?  Come on down to Publix now to get yourself a beautiful sub.` +
+        ` Just look at this beauty right here! 😍😍😍:pubsub::pubsub::pubsub:🤤🤤🤤`
+    );
+
+    embed.setImage(subData.image);
+
+    embed.setFooter(`pulled from: pubsub-api.dev.`);
+
+    return embed;
+  }
+
+  public validate(message: IMessage, args?: string[]) {
+    const input = (args || []).join('-').toLowerCase();
+
+    const keywordCheck: boolean = this._VALID_KEYS.includes(input);
+    const subTypeCheck: boolean = this._SUBS.includes(input);
+
+    return subTypeCheck || keywordCheck;
+  }
+
+  private _normalizeName(input: string): string {
```

Comment:
if you use regex you may also want to use regex to parse the command... maybe even have named fields so no more `[1]`

Reply:
the regex did not come out very pretty, but I was able to get rid of the [1] for you if that makes you happy.  Although if you still want stuff changing just let me know.